.github/workflows/test.yml

```diff
@@ -0,0 +1,122 @@
+name: test
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+  schedule:
+    - cron: "0 8 * * *"
+
+jobs:
+  pytest:
+    runs-on: ${{ matrix.os }}-latest
+    env:
+      PYTEST_ADDOPTS: "--run-integration --showlocals -vv --durations=10"
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu
+          - macos
+          - windows
+        py:
+          - '3.10'
+          - '3.9'
+          - '3.8'
+          - '3.7'
+          - '3.6'
+          - '3.5'
+          - '2.7'
+          - 'pypy3'
+          - 'pypy2'
+
+    steps:
+      - name: Setup python for tox
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install tox
+        run: python -m pip install tox
+
+      - uses: actions/checkout@v2
+
+      - name: Setup python for test ${{ matrix.py }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.py }}
```

Comment:
```suggestion
          python-version: ${{ matrix.py }}
          if: ${{ matrix.py != '3.10' }}
        - name: Setup python for test 3.10
          uses: actions/setup-python@v2
          with:
            python-version: 3.10-dev
          if: ${{ matrix.py == '3.10' }}
```

Reply:
This is not actually needed because we have an interpreter identification step. Just changing it to `3.10-dev` in the matrix works.

Reply:
Yeah, I forgot we don't pass the Python version from the matrix to tox.